hack/deployer/runner/ocp.go

```diff
@@ -198,8 +198,19 @@ func (d *OcpDriver) auth() error {
 		if err := client.ReadIntoFile(keyFileName, OcpVaultPath, OcpServiceAccountVaultFieldName); err != nil {
 			return err
 		}
-		log.Println("Setting $GCLOUD_KEYFILE_JSON...")
-		return os.Setenv("GCLOUD_KEYFILE_JSON", keyFileName)
+
+		if d.plan.Ocp.UseNonDefaultCloudSDKPath {
+			// ensure gcloud & gsutil rely on credentials stored in gcpDir instead of using the default
+			// directory (~/.config/gcloud), to not tamper any default gcloud auth already set on the system
+			log.Printf(fmt.Sprintf("Setting $CLOUDSDK_CONFIG=%s", gcpDir))
+			if err := os.Setenv("CLOUDSDK_CONFIG", gcpDir); err != nil {
+				return err
+			}
+		}
+		if err := NewCommand(fmt.Sprintf("gcloud config set project %s", d.ctx["GCloudProject"])).Run(); err != nil {
+			return err
+		}
```

Comment:
I think we can move this to the top of the function and remove the call in L226. I believe this is how it was before the previous PR.

Reply:
It should only be done after we set `$CLOUDSDK_CONFIG` otherwise we're modifying the system defaults gcloud config?

Reply:
A-ha, fair!

Would you mind adding a comment explaining why this call is here? Otherwise, someone may try to deduplicate this by doing what I suggested without knowing the consequences. :smile:

Reply:
Sure!